pulsar-functions/runtime/src/test/java/org/apache/pulsar/functions/runtime/KubernetesRuntimeFactoryTest.java

```diff
@@ -0,0 +1,172 @@
+
+        }
+
+        @Override
+        public Type getSecretObjectType() {
+            return null;
+        }
+
+        @Override
+        public void doAdmissionChecks(AppsV1Api appsV1Api, CoreV1Api coreV1Api, String jobNamespace, FunctionDetails functionDetails) {
+
+        }
+    }
+
+    private KubernetesRuntimeFactory factory;
+    private final String userJarFile;
+    private final String pulsarRootDir;
+    private final String javaInstanceJarFile;
+    private final String pythonInstanceFile;
+    private final String pulsarServiceUrl;
+    private final String pulsarAdminUrl;
+    private final String stateStorageServiceUrl;
+    private final String logDirectory;
+
+    public KubernetesRuntimeFactoryTest() throws Exception {
+        this.userJarFile = "UserJar.jar";
+        this.pulsarRootDir = "/pulsar";
+        this.javaInstanceJarFile = "/pulsar/instances/java-instance.jar";
+        this.pythonInstanceFile = "/pulsar/instances/python-instance/python_instance_main.py";
+        this.pulsarServiceUrl = "pulsar://localhost:6670";
+        this.pulsarAdminUrl = "http://localhost:8080";
+        this.stateStorageServiceUrl = "bk://localhost:4181";
+        this.logDirectory = "logs/functions";
+    }
+
+    @AfterMethod
+    public void tearDown() {
+        if (null != this.factory) {
+            this.factory.close();
+        }
+    }
+
+    KubernetesRuntimeFactory createKubernetesRuntimeFactory(String extraDepsDir) throws Exception {
+        KubernetesRuntimeFactory factory = spy(new KubernetesRuntimeFactory(
+            null,
+            null,
+            null,
+            pulsarRootDir,
+            false,
+            true,
+            "myrepo",
+            "anotherrepo",
+            extraDepsDir,
+            null,
+            pulsarServiceUrl,
+            pulsarAdminUrl,
+            stateStorageServiceUrl,
+            null,
+            null,
+            null,
+            null,
+            new TestSecretProviderConfigurator()));
+        doNothing().when(factory).setupClient();
+        return factory;
+    }
+
+    FunctionDetails createFunctionDetails() {
+        FunctionDetails.Builder functionDetailsBuilder = FunctionDetails.newBuilder();
+        functionDetailsBuilder.setRuntime(FunctionDetails.Runtime.JAVA);
+        functionDetailsBuilder.setTenant("public");
+        functionDetailsBuilder.setNamespace("default");
+        functionDetailsBuilder.setName("function");
+        functionDetailsBuilder.setSecretsMap("SomeMap");
+        return functionDetailsBuilder.build();
+    }
+
+    @Test
+    public void testAdmissionChecks() throws Exception {
+        factory = createKubernetesRuntimeFactory(null);
+        FunctionDetails functionDetails = createFunctionDetails();
+        factory.doAdmissionChecks(functionDetails);
+        verify(factory, times(1)).setupClient();
```

Comment:
we should probably also check if:

AppsV1Api appsClient;
CoreV1Api coreClient;

is not null.

Reply:
setup client will bail out if they cannot be init, so that will not be the right check